scarches/models/scgen/vaearith_model.py

```diff
@@ -0,0 +1,344 @@
+        """
+        #device = next(self.model.parameters()).device # get device of model.parameters
+        #adata_tensor = torch.tensor(adata, device=device) # to tensor
+        output = self.model.predict(adata, conditions, cell_type_key, condition_key, adata_to_predict, celltype_to_predict, obs_key)
+        prediction = output[0].cpu().detach()
+        delta = output[1].cpu().detach()
+        return np.array(prediction), np.array(delta)
+
+
+    def batch_removal(self, adata, batch_key, cell_label_key, return_latent = True):
+        """
+        Removes batch effect of adata
+
+        Parameters
+        ----------
+        adata: `~anndata.AnnData`
+            Annotated data matrix. adata must have `batch_key` and `cell_label_key` which you pass to the function in its obs.
+        batch_key: `str`
+            batch label key in  adata.obs
+        cell_label_key: `str`
+            cell type label key in adata.obs
+        return_latent: `bool`
+            if `True` returns corrected latent representation
+
+        Returns
+        -------
+        corrected: `~anndata.AnnData`
+            adata of corrected gene expression in adata.X and corrected latent space in adata.obsm["latent_corrected"].
+        """
+        corrected = self.model.batch_removal(adata, batch_key, cell_label_key, return_latent)
+        return corrected
+
+
+    @classmethod
+    def _validate_adata(cls, adata, dct):
+        if adata.n_vars != dct['x_dim_']:
+            raise ValueError("Incorrect var dimension")
+
+    @classmethod
+    def _get_init_params_from_dict(cls, dct):
+        init_params = {
+            'hidden_layer_sizes': dct['hidden_layer_sizes_'],
+            'z_dimension': dct['z_dim_'],
+            'dr_rate': dct['dr_rate_'],
+        }
+
+        return init_params
+
+
+    @classmethod
+    def map_query_data(cls, corrected_reference: AnnData, query: AnnData, reference_model: Union[str, 'scgen'], return_latent = True):
+        """
+        Removes the batch effect between reference and query data.
+        Additional training on query data is not needed.
+
+        Parameters
+        ----------
+        corrected_reference: `~anndata.AnnData`
+           Already corrected reference anndata object
+        query: `~anndata.AnnData`
+            Query anndata object
+        batch_key: `str`
+            batch label key in  adata.obs
+        cell_label_key: `str`
+            cell type label key in adata.obs
+        return_latent: `bool`
+            if `True` returns corrected latent representation
+
+        Returns
+        -------
+        integrated: `~anndata.AnnData`
+        Returns an integrated query.
+        """
+        if isinstance(reference_model, str):
+            attr_dict, model_state_dict, var_names = cls._load_params(reference_model)
+            _validate_var_names(query, var_names)
+        else:
+            attr_dict = reference_model._get_public_attributes()
```

Comment:
There is no point in reinitialization of the model, we don't expand anything. Just use the model if it is passed:
if isinstance(reference_model, str):
  load the model as new_model
else:
  new_model = reference_model